engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/PendingEdits.scala

```diff
@@ -20,6 +21,20 @@ trait PendingEdits {
     */
   def dequeue(file: File): Seq[PendingEdit]
 
+  /** Put the IdMap of the file.
+    *
+    * @param file the edited file
+    * @param idMap the new IdMap
+    */
+  def putIdMap(file: File, idMap: IdMap): Unit
+
+  /** Consume the idMap of the file.
+    *
+    * @param file the edited file
+    * @return the IdMap of the file
+    */
+  def takeIdMap(file: File): Option[IdMap]
```

Comment:
`removeIdMap` or `getAndRemove`. `take` doesn't really say that we are removing the association.